tests/integration/targets/test_azure_virtual_machine_with_public_ip/tasks/delete_all_resources.yml

```diff
@@ -0,0 +1,92 @@
+---
+- name: Delete Virtual Machine and all associated resources
+  azure.azcollection.azure_rm_virtualmachine:
+    resource_group: "{{ vm_resource_group }}"
+    name: "{{ vm_name }}"
+    state: absent
+  retries: 40
+  delay: 5
+  register: result
+  until: result.failed == false
+
+- name: Delete Availability set
+  azure.azcollection.azure_rm_availabilityset:
+    resource_group: "{{ vm_resource_group }}"
+    name: "{{ availability_set }}"
+    state: absent
+  retries: 40
```

Comment:
Do you actualy need each resource deletion to have retries? isn't it enough have the retries only in the first deletion?

Reply:
I thought about it, but if resource deleted or doesn't exist it doesn't try 40 times, so it won't hurt